olive/passes/onnx/model_builder.py

```diff
@@ -184,15 +174,26 @@ def _run_for_config(
             if config[arg] is not None:
                 extra_args[arg] = "1" if config[arg] else "0"
 
-        create_model(
-            model_name=model_path,
-            input_path=input_path,
-            output_dir=str(output_model_filepath.parent),
-            precision=precision,
-            execution_provider=target_execution_provider,
-            cache_dir=cache_dir,
-            **extra_args,
-        )
+        try:
+            create_model(
+                model_name=model_path,
+                input_path=input_path,
+                output_dir=str(output_model_filepath.parent),
+                precision=precision,
+                execution_provider=target_execution_provider,
+                # model builder uses the cache_dir both as hf cache and also to store intermediate files
+                # not ideal, but we can't change this without changing the model builder
+                cache_dir=transformers.utils.TRANSFORMERS_CACHE,
+                **extra_args,
+            )
+        except Exception:
+            # if model building fails, clean up the intermediate files in the cache_dir
+            cache_dir = Path(transformers.utils.TRANSFORMERS_CACHE)
+            if cache_dir.is_dir():
+                for file in cache_dir.iterdir():
+                    if file.suffix == ".bin":
+                        file.unlink()
```

Comment:
Do we really want to do this? Transformer cache may contain data from other models as well. Blindly deleting everything doesn't seem like a good idea.

Reply:
Not really needed but it cleans up the cache directory if the model builder fails for some reason. I checked the cache structure and hf cache creates subdirectories inside it for models, datasets, etc. We are only iterating over the top level where the .bin files are generated by model builder. It is the same logic the model builder does when it succeeds https://github.com/microsoft/onnxruntime-genai/blob/c7eba3c63a454edd6662eb007ff397d1146cc081/src/python/py/models/builder.py#L378

Reply:
for reference
![image](https://github.com/user-attachments/assets/8021ee91-eb9c-4e8b-a5a8-1545cd26eafd)